subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/JavaApplicationInitIntegrationTest.groovy

```diff
@@ -36,10 +36,41 @@ class JavaApplicationInitIntegrationTest extends AbstractInitIntegrationSpec {
 
     def "defaults to Groovy build scripts"() {
         when:
-        run('init', '--type', 'java-application')
+        run (tasks)
 
         then:
         dslFixtureFor(GROOVY).assertGradleFilesGenerated()
+
+        where:
+        tasks << [['init', '--type', 'java-application'],
+                  ['init', '--type', 'java-application', '--incubating']]
+    }
```

Comment:
Maybe annotate with `@Unroll` and add a `#` variable name in the test name?

Since the two states you are testing are so similar, I might rewrite as:

```
def 'defaults to Groovy build scripts when incubating flag=#isIncubating'() {
  ...
  when:
  run(['init', '--type', 'java-application', isIncubating ? '--incubating' : null])
  ...
  where:
  isIncubating << [true, false]
}

```

Reply:
Don't change it back, but IMHO the explicit list is easier to understand than a boolean in a test.

We could have done this to make it clearer what the test was doing:
```
def "defaults to Groovy build scripts #tasks"() {
```
A lot of the time, a boolean in a test screams "there are two different tests here" whereas two sets of arguments feel like the same test. But six of one, half a dozen of the other. The latest changes are OK.